src/audiences/ui/list.js

```diff
@@ -316,11 +325,13 @@ const applyWithSelect = withSelect( select => {
 		getIsLoading,
 		getPagination,
 	} = select( 'audience' );
-	const posts = getPosts();
+	const canCreate = select( 'core' ).canUser( 'create', 'audiences' );
 	const loading = getIsLoading();
 	const pagination = getPagination();
-
-	const canCreate = select( 'core' ).canUser( 'create', 'audiences' );
+	const posts = canCreate ? getPosts() : getPosts( {
+		context: 'view',
+		status: 'publish',
+	} );
```

Comment:
Ternary conditions tend to work best for shorter values, sometimes they have poor readability. It's a judgment call ultimately but for this logic I would probably write it like this in keeping with how you've done it elsewhere in this file.

```suggestion
	const queryArgs = {
		context: canCreate ? 'edit' : 'view',
		status: canCreate ? 'publish,draft' : 'publish',
	};
	const posts = getPosts( queryArgs );
```

What you could do is write a small helper function to get these params like:

```js
const getQueryArgs = ( queryArgs = {}, shouldOverride = false, overrides = {} ) => {
  return Object.assign(
    {
      context: 'view',
      status: 'publish',
    },
    queryArgs,
    shouldOverride ? overrides : {}
  );
}
```

Then use it like

```js
const posts = getPosts( getQueryArgs( {}, canCreate, {
  context: 'edit',
  status: 'publish,draft'
} ) );
```

Something like that anyway. It's not a blocker for me as the use case is limited to just this file so don't feel like you have to do that.

Reply:
I think the suggested change probably looks cleaner. I don't think we need a helper function at this point, but I've been wondering if something like it might be useful as I was going through these changes. I think it matters if this is something we come back to in the future and start using the query args more.